<launch>

      <node pkg="hku_m100_landing" type="position_track_velocity_controller" name="position_track_velocity_controller_node" output="screen">
      </node>

    <node ns="hku_m100_landing" name="position_track_pid_velocity_x_controller" pkg="pid" type="controller">
      <param name="node_name" value="apriltag_tracking_x_controller" />
<!--       <param name="Kp" value="0.8" />
      <param name="Ki" value="0.3" />
      <param name="Kd" value="300" /> -->
      <param name="Kp" value="0.9" />
      <param name="Ki" value="0.3" />
      <param name="Kd" value="50" />
      <param name="P" value="1" />
      <param name="Q" value="0.00005" />
      <param name="R" value="1" />
      <param name="/use_sim_time" value="false" />
      <param name="converge_threshold" value="0.05" />
      <param name="upper_limit" value="15" />
      <param name="lower_limit" value="-15" />
      <param name="windup_limit" value="20" />
      <param name="cutoff_frequency" value="20" />
      <param name="diagnostic_period" value="0.25" />
      <param name="max_loop_frequency" value="105.0" />
      <param name="min_loop_frequency" value="95.0" />
      <remap from="setpoint" to="/teamhku/position_track/setpoint_x" />
      <param name="topic_from_plant" value="/teamhku/position_track/x_state" />
      <param name="topic_from_controller" value="/teamhku/position_track/velocity_control_effort_x" />
     </node>

    <node ns="hku_m100_landing" name="position_track_pid_velocity_y_controller" pkg="pid" type="controller"  >
      <param name="node_name" value="apriltag_tracking_y_controller" />
<!--       <param name="Kp" value="0.8" />
      <param name="Ki" value="0.3" />
      <param name="Kd" value="300" /> -->
      <param name="Kp" value="0.9" />
      <param name="Ki" value="0.3" />
      <param name="Kd" value="50" />
      <param name="P" value="1" />
      <param name="Q" value="0.00005" />
      <param name="R" value="1" />
      <param name="converge_threshold" value="0.05" />
      <param name="upper_limit" value="15" />
      <param name="lower_limit" value="-15" />
      <param name="windup_limit" value="20" />
      <param name="cutoff_frequency" value="20" />
      <param name="diagnostic_period" value="0.25" />
      <param name="max_loop_frequency" value="105.0" />
      <param name="min_loop_frequency" value="95.0" />
      <remap from="setpoint" to="/teamhku/position_track/setpoint_y" />
      <param name="topic_from_plant" value="/teamhku/position_track/y_state" />
      <param name="topic_from_controller" value="/teamhku/position_track/velocity_control_effort_y" />
    </node>
    
    <node name="position_track_setpoint" pkg="hku_m100_landing" type="position_track_setpoint" output="screen" >
      <param name="tag_36h11_detection_topic" value="/apriltags/36h11/detections" />
      <param name="tag_16h5_detection_topic" value="/apriltags/16h5/detections" />
      <param name="landing_threshold_36h11" value="0.3" />
      <param name="landing_threshold_16h5" value="0.1" />
      <param name="reland_height_min_threshold" value="1.5" />
      <param name="reland_height_max_threshold" value="2.5" />
    </node>


<!-- 
    <node ns="hku_m100_landing" name="position_track_pid_yaw_controller" pkg="pid" type="controller" >
      <param name="node_name" value="apriltag_tracking_yaw_controller" />
      <param name="Kp" value="0.0" />
      <param name="Ki" value="0.0" />
      <param name="Kd" value="0.0" />
      <param name="converge_threshold" value="1" />
      <param name="upper_limit" value="10" />
      <param name="lower_limit" value="-10" />
      <param name="windup_limit" value="5" />
      <param name="cutoff_frequency" value="20" />
      <param name="diagnostic_period" value="0.25" />
      <param name="max_loop_frequency" value="105.0" />
      <param name="min_loop_frequency" value="95.0" />
      <remap from="setpoint" to="/teamhku/position_track/setpoint_yaw" />
      <param name="topic_from_plant" value="/teamhku/position_track/yaw_state" />
      <param name="topic_from_controller" value="/teamhku/position_track/velocity_control_effort_yaw" />
     </node> -->

</launch>
